storage/entity.go

```diff
@@ -0,0 +1,350 @@
+package storage
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"io/ioutil"
+
+	"github.com/satori/uuid"
+)
+
+// Annotating as secure for gas scanning
+/* #nosec */
+const (
+	partitionKeyNode  = "PartitionKey"
+	rowKeyNode        = "RowKey"
+	etagErrorTemplate = "Etag didn't match: %v"
+)
+
+// Entity represents an entity inside an Azure table.
+type Entity struct {
+	tsc           *TableServiceClient
```

Comment:
What do you think about exporting `Table` field? That way it is easy to change the Table in which the entity operations are performed.
For using different storage accounts, until now the thing to do is create a new client. Adding the functionality to easily change storage accounts I think is better suited for a different PR.
About the client and table as fields, we want to follow a similar approach to what the [C# SDK](https://github.com/Azure/azure-storage-net/blob/master/Samples/GettingStarted/VisualStudioQuickStarts/DataFileStorage/Program.cs) is doing.